pkg/client/register.go

```diff
@@ -39,13 +38,8 @@ import (
 // Error Handling:
 //   - If the RegisterClient method of the services.RegistrationService encounters any error, the method will return
 //     a detailed error message wrapped with the context of client registration failure.
-func RegisterClient(
-	name string,
-	grantTypes []types.GrantTypeEnum,
-	redirectURIs []string,
-	clientType types.ClientTypeEnum,
-	db database.Database,
-) (*models.RegistrationResponse, error) {
+func RegisterClient(name string, grantTypes []models.GrantTypeEnum, redirectURIs []string, clientType models.TypeEnum) (*models.RegistrationResponse, error) {
+	db := config.GetDatabase()
```

Comment:
If the user didn't previously connect a database to the library, this will default to the `InMemoryDatabase`